certbot-apache/certbot_apache/configurator.py

```diff
@@ -884,65 +921,77 @@ def _copy_create_ssl_vhost_skeleton(self, avail_fp, ssl_fp):
 
         try:
             with open(avail_fp, "r") as orig_file:
-                with open(ssl_fp, "w") as new_file:
-                    new_file.write("<IfModule mod_ssl.c>\n")
-
-                    comment = ("# Some rewrite rules in this file were "
-                              "disabled on your HTTPS site,\n"
-                              "# because they have the potential to create "
-                              "redirection loops.\n")
+                orig_file_list = [line for line in orig_file]
+                if vhost_num != -1:
+                    orig_file_list = self._create_block_segments(orig_file_list, vhost_num)
 
-                    for line in orig_file:
-                        A = line.lstrip().startswith("RewriteCond")
-                        B = line.lstrip().startswith("RewriteRule")
-
-                        if not (A or B):
-                            new_file.write(line)
-                            continue
+            if ssl_fp in self._skeletons:
+                bit = "a"
+                self._skeletons[ssl_fp].append(avail_fp)
+            else:
+                bit = "w"
+                self._skeletons[ssl_fp] = [avail_fp]
+
+            with open(ssl_fp, bit) as new_file:
```

Comment:
Why `a+`? Given that the options are currently either appending (`a`) to a file or writing it from scratch (`w`), why would adding the ability to read (`+`) help to unify them?

It seems to me that `a` *might* work for both cases, but `w` is a potentially (?) safer because it will overwrite anything weird that's in the file that doesn't match `self._skeletons`' idea of what should be there. Note however that this is my understanding from looking at this function in isolation, without a well-formed idea of context.